test/solver/gmres_kernels.cpp

```diff
@@ -305,11 +313,35 @@ TEST_F(Gmres, GmresApplyOneRHSIsEquivalentToRef)
     auto exec_solver = exec_gmres_factory->generate(d_mtx);
     auto b = gen_mtx(m, n);
     auto x = gen_mtx(m, n);
-    auto d_b = gko::clone(exec, b);
-    auto d_x = gko::clone(exec, x);
+    auto d_b = Mtx::create(exec);
+    auto d_x = Mtx::create(exec);
+    d_b->copy_from(b.get());
+    d_x->copy_from(x.get());
```

Comment:
clone?

Reply:
`clone` does not give you the same type, you only get a `LinOp` from it.

Reply:
Are you sure about that? b and x are `Mtx` types, so they implement `clone` via `EnableLinOp` returning their actual type, IIRC.

Reply:
My bad, both of you are correct. Using `gko::clone` uses the actual type and not just `LinOp` (See [declaration of `gko::clone()`](https://github.com/ginkgo-project/ginkgo/blob/develop/include/ginkgo/core/base/utils_helper.hpp#L186)).
I will change it here and see if this change was done somewhere else as well.